src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DhGaugeParameters.hpp

```diff
@@ -0,0 +1,153 @@
+  using type = double;
+  static std::string name() noexcept { return "RollOnTimeWindow"; }
+  static constexpr Options::String help{
+      "The width of the Gaussian that controls how quickly the gauge is "
+      "rolled on."};
+  using group = gauges::OptionTags::GaugeGroup;
+};
+
+/// The width of the Gaussian for the spatial decay of the damped harmonic
+/// gauge.
+template <typename Frame>
+struct DampedHarmonicSpatialDecayWidth {
+  using type = double;
+  static std::string name() noexcept { return "SpatialDecayWidth"; }
+  static constexpr Options::String help{
+      "Spatial width of weight function used in the damped harmonic "
+      "gauge."};
+  using group = gauges::OptionTags::GaugeGroup;
+};
+
+/// The amplitudes for the L1, L2, and S terms, respectively, for the damped
+/// harmonic gauge.
+template <typename Frame>
+struct DampedHarmonicAmplitudes {
+  using type = std::array<double, 3>;
+  static std::string name() noexcept { return "Amplitudes"; }
+  static constexpr Options::String help{
+      "Amplitudes [AL1, AL2, AS] for the damped harmonic gauge."};
+  using group = gauges::OptionTags::GaugeGroup;
+};
+
+/// The exponents for the L1, L2, and S terms, respectively, for the damped
+/// harmonic gauge.
+template <typename Frame>
+struct DampedHarmonicExponents {
+  using type = std::array<int, 3>;
+  static std::string name() noexcept { return "Exponents"; }
+  static constexpr Options::String help{
+      "Exponents [eL1, eL2, eS] for the damped harmonic gauge."};
+  using group = gauges::OptionTags::GaugeGroup;
+};
+}  // namespace OptionTags
+
+/*!
+ *  A struct holding the parameters for initializing damped harmonic gauge
+ *  If UseRollon is true, the gauge transitions ("rolls on") to damped harmonic
+ *  fromdetermines whether the gauge transitions to damped harmonic
+ * ("rolls on") from the initial data gauge; otherwise, the gauge begins
+ * immediately in damped harmonic gauge.
+ */
+template <typename Frame, bool UseRollon>
+struct DhGaugeParameters;
+
+template <typename Frame>
+struct DhGaugeParameters<Frame, true> {
+  double rollon_start;
+  double rollon_window;
+  double spatial_decay_width;
+  std::array<double, 3> amplitudes;
+  std::array<int, 3> exponents;
+
+  static constexpr Options::String help{
+      "A struct holding the parameters for initializing damped harmonic "
+      "gauge, including a roll-on from the initial gauge."};
+
+  using options = tmpl::list<OptionTags::DampedHarmonicRollOnStart,
+                             OptionTags::DampedHarmonicRollOnWindow,
+                             OptionTags::DampedHarmonicSpatialDecayWidth<Frame>,
+                             OptionTags::DampedHarmonicAmplitudes<Frame>,
+                             OptionTags::DampedHarmonicExponents<Frame>>;
+
+  DhGaugeParameters(const double start, const double window, const double width,
+                    const std::array<double, 3>& amps,
+                    const std::array<int, 3>& exps)
```

Comment:
If we can remove then `Frame` template parameter, could you move the constructions and pup functions into a `cpp` file please?